core/src/test/scala/org/apache/spark/SparkFunSuite.scala

```diff
@@ -52,6 +62,23 @@ abstract class SparkFunSuite
     getTestResourceFile(file).getCanonicalPath
   }
 
+  private def saveThreadNames(): Unit = {
```

Comment:
Suggest turning this into runningThreadNames(): Set[String], and then you can use this method both in beforeAll() and in printRemainingThreadNames() (line 70). And you can maybe put the whitelist logic here as well.

Reply:
Config file removed and refactored as you suggested. It's much more simple now, thanks :)